src/components/SidePanel/SidePanel.test.js

```diff
@@ -0,0 +1,203 @@
+import React from 'react'
+import { render, fireEvent } from '@testing-library/react'
+import user from '@testing-library/user-event'
+import SidePanel from './'
+
+jest.useFakeTimers()
+
+describe('Show / no show', () => {
+  test('should be shown or hidden using show prop using useAnimatedRender under the hood', () => {
+    const { queryByTestId, rerender } = render(<SidePanel />)
+
+    expect(queryByTestId('sidepanel-overlay')).not.toBeInTheDocument()
+
+    rerender(<SidePanel show />)
+    fireEvent.animationEnd(queryByTestId('sidepanel-overlay'))
+
+    expect(queryByTestId('sidepanel-overlay')).toBeInTheDocument()
+    expect(
+      queryByTestId('sidepanel-overlay').classList.contains('element-in')
+    ).toBe(true)
+    expect(
+      queryByTestId('sidepanel-overlay').querySelector('.SidePanel')
+    ).toBeInTheDocument()
+    expect(document.activeElement).toBe(
+      queryByTestId('sidepanel-overlay').querySelector('.SidePanel')
+    )
+
+    rerender(<SidePanel show={false} />)
+    fireEvent.animationEnd(queryByTestId('sidepanel-overlay'))
+
+    expect(queryByTestId('sidepanel-overlay')).not.toBeInTheDocument()
+  })
+})
+
+describe('Contents', () => {
+  test('Header', () => {
+    const { container } = render(<SidePanel show />)
+    const header = container.querySelector('.SidePanel__Header')
+
+    expect(header).toBeInTheDocument()
+    expect(header.tagName).toBe('HEADER')
+
+    const heading = header.querySelector('h1')
+
+    expect(heading).toBeInTheDocument()
+    expect(heading.getAttribute('id')).toBe('sidepanel-header-heading')
+    expect(heading.textContent).toBe('Review and Start')
```

Comment:
Similar to above, in those 2 lines we could use `toHaveAttribute` and `toHaveTextContent` as well. And the same could be applied to some tests below as well.

Reply:
ditto